tests/admin_views/tests.py

```diff
@@ -6166,3 +6167,70 @@ def test_implicitly_generated_pk(self):
         post_data = {'name': '2'}
         response = self.client.post(reverse('admin:admin_views_implicitlygeneratedpk_change', args=(1,)), post_data)
         self.assertEqual(response.status_code, 302)
+
+
+@override_settings(
+    PASSWORD_HASHERS=['django.contrib.auth.hashers.SHA1PasswordHasher'],
+    ROOT_URLCONF="admin_views.urls",
+)
+class SiteEachContextTest(TestCase):
+    """
+    Check each_context contains the documented variables, check that available_apps context
+    variable structure is the expected one
+    """
+    @classmethod
+    def setUpTestData(cls):
+        cls.u1 = User.objects.create(
+            id=100, password='sha1$995a3$6011485ea3834267d719b4c801409b8b1ddd0158',
+            last_login=datetime.datetime(2007, 5, 30, 13, 20, 10), is_superuser=True, username='super',
+            first_name='Super', last_name='User', email='super@example.com',
+            is_staff=True, is_active=True, date_joined=datetime.datetime(2007, 5, 30, 13, 20, 10)
+        )
+
+    def setUp(self):
+        factory = RequestFactory()
+        request = factory.get(reverse('admin6:index'))
+        request.user = self.u1
+        self.ctx = site6.each_context(request)
+
+    def test_each_context(self):
+        ctx = self.ctx
+        self.assertIn('site_header', ctx)
```

Comment:
we can remove the `assertIn` assertions I think. If we get a `KeyError` on the `assertEqual` assertions, that's just as helpful.